### ISSUE TITLE
Help broke

### ISSUE THREAD

**Issue by GregTheGreek** [tj/commander.js#1092]
It seems help is broken when using `reuqiredOption`

```
$ node index.js --help
error: required option '-t, --target <url>' not specified
```

**Comment by shadowspawn** [comment]
Yes, the requiredOption is being enforced before the help is displayed which is a bug. (As a work-around you can get to the help by supplying the required option.)

Already reported in #1089 
Fix on the way: #1091

**Comment by shadowspawn** [comment]
Should be fixed in 4.0.1 which has been released.
